pkgs/applications/emulators/dolphin-emu/primehack.nix

```diff
@@ -1,138 +1,164 @@
 { lib
 , stdenv
 , fetchFromGitHub
-, pkg-config
+, fetchpatch
 , cmake
+, pkg-config
 , wrapQtAppsHook
-, qtbase
+, alsa-lib
 , bluez
+, bzip2
+, cubeb
+, curl
+, enet
 , ffmpeg
-, libao
-, libGLU
+, fmt_10
+, gtest
+, hidapi
+, libevdev
 , libGL
-, pcre
-, gettext
-, libXrandr
+, libiconv
+, libpulseaudio
+, libspng
 , libusb1
-, libpthreadstubs
-, libXext
-, libXxf86vm
-, libXinerama
-, libSM
 , libXdmcp
-, readline
-, openal
-, udev
-, libevdev
-, portaudio
-, curl
-, alsa-lib
-, miniupnpc
-, enet
+, libXext
+, libXrandr
+, lz4
+, lzo
 , mbedtls_2
-, soundtouch
+, miniupnpc
+, openal
+, pugixml
+, qtbase
+, qtsvg
 , sfml
-, fmt
-, xz
+, udev
 , vulkan-loader
-, libpulseaudio
+, xxHash
+, xz
 
-# - Inputs used for Darwin
+  # Darwin-only dependencies
 , CoreBluetooth
 , ForceFeedback
+, IOBluetooth
 , IOKit
+, moltenvk
 , OpenGL
-, libpng
-, hidapi
+, VideoToolbox
 }:
 
 stdenv.mkDerivation rec {
   pname = "dolphin-emu-primehack";
-  version = "1.0.6a";
+  version = "1.0.7a";
 
   src = fetchFromGitHub {
     owner = "shiiion";
     repo = "dolphin";
     rev = version;
-    sha256 = "sha256-gc4+ofoLKR+cvm+SaWEnGaKrSjWMKq7pF6pEIi75Rtk=";
+    sha256 = "sha256-vuTSXQHnR4HxAGGiPg5tUzfiXROU3+E9kyjH+T6zVmc=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # fix build w/ glibc-2.39
+    (fetchpatch {
+      url = "https://github.com/dolphin-emu/dolphin/commit/3da2e15e6b95f02f66df461e87c8b896e450fdab.patch";
+      hash = "sha256-+8yGF412wQUYbyEuYWd41pgOgEbhCaezexxcI5CNehc=";
+    })
+  ];
+
   nativeBuildInputs = [
-    pkg-config
+    stdenv.cc
     cmake
-  ] ++ lib.optional stdenv.isLinux wrapQtAppsHook;
+    pkg-config
+    wrapQtAppsHook
+  ];
 
-  buildInputs = [
+  buildInputs = lib.optionals stdenv.isDarwin [
```

Comment:
optional arguments usually go after the none optional ones

Reply:
I'll take the opportunity to blame #238959 again (and because I copied from regular Dolphin).